app/services/MatcherPool.scala

```diff
@@ -69,8 +69,8 @@ class MatcherPool(
   val checkTimeoutDuration: FiniteDuration = 5 seconds
 )(implicit ec: ExecutionContext, implicit val mat: Materializer) extends Logging {
   type JobProgressMap = Map[String, Int]
-
-  private val matchers = new ConcurrentHashMap[String, (Category, Matcher)]().asScala
+  type MatcherId = String
+  private val matchers = new ConcurrentHashMap[MatcherId, Matcher]().asScala
```

Comment:
NB: instead of mapping from a category to a matcher, we give matchers a unique ID and use that as the key. The filtering code in `runValidationJob` below determines which matches are eligible for which jobs.